sdk/trace/sampling_test.go

```diff
@@ -245,3 +245,50 @@ func TestTracestateIsPassed(t *testing.T) {
 		})
 	}
 }
+
+func TestAlwaysRecordRootSampled(t *testing.T) {
```

Comment:
Those two tests are very similar. How about using subtests?
Also, you do test that `Drop` becomes a `Record`, and that `RecordAndSample` doesn't change. But not testing that `Record` remains `Record`.